engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,102 @@
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            return org.system_purposes.addons.length;
+        });
+
+        $scope.purposeAddonsList = function () {
+            if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
+                $scope.defaultOptions = ['No Change', 'None (Clear)'];
+                $scope.addons = $scope.organization.system_purposes.addons.map(function (addon) {
+                    return addon; });
+                $scope.addonList = $scope.defaultOptions.concat($scope.addons);
+                return $scope.addonList;
+            }
+            return [];
+        };
+
+        $scope.defaultUsages = ['No change', 'None (Clear)', 'Production', 'Development/Test', 'Disaster Recovery'];
+        $scope.defaultRoles = ['No change', 'None (Clear)', 'Red Hat Enterprise Linux Server', 'Red Hat Enterprise Linux Workstation', 'Red Hat Enterprise Linux Compute Node'];
+        $scope.defaultServiceLevels = ['No change', 'None (Clear)', 'Self-Support', 'Standard', 'Premium'];
+
+        $scope.hostCount = hostIds.included.ids.length;
+
+        $scope.selectedUsages = $scope.defaultUsages[0];
+        $scope.selectedRoles = $scope.defaultRoles[0];
+        $scope.selectedServiceLevels = $scope.defaultServiceLevels[0];
+
+        $scope.selected = {
+           addons: undefined
+        };
+
+        $scope.selectedItemToParam = function (item) {
+            var mapping = {
+                "None (Clear)": "",
+                "No change": null
+            };
+            if (mapping.hasOwnProperty(item)) {
+                return mapping[item];
+            }
+            return item;
+        };
+
+        function actionParams() {
+            var params = hostIds;
+
+            params['purpose_usage'] = $scope.selectedItemToParam($scope.selectedUsages);
+            params['purpose_role'] = $scope.selectedItemToParam($scope.selectedRoles);
+            params['purpose_addons'] = $scope.selectedItemToParam($scope.selectedAddons);
+            params['service_level'] = $scope.selectedItemToParam($scope.selectedServiceLevels);
+
+            if (angular.isUndefined(params['purpose_addons'])) {
+                params['purpose_addons'] = null;
+            } else if (params['purpose_addons'].includes('No Change')) {
+                params['purpose_addons'] = null;
+            } else if (params['purpose_addons'] === "") {
+                params['purpose_addons'] = [];
+            }
+            
+            if (params['purpose_addons'] && params['purpose_addons'].includes('None (Clear)') && params['purpose_addons'].length > 1) {
```

Comment:
Per our discussion the other day, since it's possible for the user to make illogical selections, we want to cover the following situations:
1. If both 'No change' and other addon items are selected, 'No change' wins
2. If 'No change' and 'None' are both selected, 'No change' wins 
3. If both 'None (Clear)' and others are selected, None is ignored and the others win

After thinking about it for a while, I arrived at an alternate way of thinking about it.  

#### Input

You have a multi-select with several options:
* No Change
* None (Clear)
* addon1
* addon2
* addon3

The first two options are what I would call possible user _intents_, whereas the rest of the options are actual selected addons.  

#### Outputs

We want to end up with a param that looks like one of the following:
* `[]` for `None (Clear)`
* `null` for `No Change`
* Otherwise, the actual array of add-ons

So I think it makes sense to add a `$scope.selectedAddonsToParam` which will do the following:

1. Divide the state of the checkbox into _intents_ and _selectedAddons_
2. Figure out what the user _intent_ is;
3. Use that to figure out which of the above three to return.

This would accomplish the same thing as the if-else blocks here.

We could modify `$scope.selectedItemToParam` to check for arrays and also call our new method:
```js
$scope.selectedItemToParam = function (item) {
          console.log({item})
            var mapping = {
                "None (Clear)": "",
                "No change": null,
                "": [],
            };
            if (Array.isArray(item)) {
              return $scope.selectedAddonsToParam(item);
            }
            if (mapping.hasOwnProperty(item)) {
                return mapping[item];
            }
            return item;
        };
```